### DISCUSSION TITLE
Creating email queue

### DISCUSSION THREAD

**Discussion by MikeKry** [OrchardCMS/OrchardCore#11129] [Q&A] [answered]
Hi everyone,

I would like to implement an email queue for a project and it does not seem to exist in OrchardCore. 
It would log every email that should be sent and in case of error (e.g. SMPT misconfiguration) it would keep the mail in queue and try to resend manually or by scheduled task. 

I am interested in two things.
1) is someone already trying to do that for OC? I searched github and google and did not find anything, only something named Jobs Queue from OrchardCMS (seems that it is being implemented also for OCore). but that also does not cover this use case (resend in case of failure) 
2) if I will be implementing this feature, how should I manage it in database? Storing it in Document table would create a lot of rows every day in some use cases and I am still not sure what happens if the DocumentID row overflows its limit which could be a real scenario if everything is stored in one table? Is it better to store data like this in separate table? 

Thanks for answers :)

**Comment by hishamco** [comment]
So summarize things @deanmarcussen is already did work in Job Queues and demo it in Orchard Core Standup, also he is WIP for Scheduling Infrastructure APIs

**Reply by MikeKry** [reply]
@hishamco 

As I mentioned in my post - I know that, but Job queue is not what I want...

Difference of email queue against job queue:
- it creates log of send mail with their status
- in case of failure, it tries to resend up to defined limit (10 times for example) in case SMTP is unavailable or something
- in case of persitent failure, it stops sending failed mails, but keeps them for manual inspection
- primary purpose is to not lose any mail due to technical difficulties, not to schedule them in some queue
- execution unit is not a job but an email

Anyway, if there is nothing else like this, then I will create it myself, but I am still curious about database side (second question)


just to be clear - I think this is custom module that does not have to be added to OrchardCore (even though some other CMS do have this) - so I am not asking for anyone to implement it, just wanted to check if I did not overlook anything existing and for some guidance. I should be able to implement it with background tasks, but not sure about db structure.

**Reply by hishamco** [reply]
Nothing in OC for what you want but you can exten the `OrchardCore.Email` module with the help of background tasks that's available too

**Comment by hishamco** [comment]
So summarize things @deanmarcussen is already did work in Job Queues and demo it in Orchard Core Standup, also he is WIP for Scheduling Infrastructure APIs

**Comment by MikeKry** [comment]
Ok, I have found an issue in @sebastienros yessql repo about same theoretical issue but not resolved - https://github.com/sebastienros/yessql/issues/142

Maybe I should ask there if it is a good practice to place big - unrelated to CMS - data into custom table instead of Document table

**Comment by MikeKry** [comment, marked as answer]
In case anyone ends here - answer to my question is using collections - implemented in https://github.com/sebastienros/yessql/pull/30
